File: version1/backend/client/components/Product.js
import React from "react";
import { Card } from "react-bootstrap";
import Rating from "./Rating";
import Link from "next/link";

const Product = ({ product }) => {
  return (
    <Card className="my-3 p-3 rounded">
      <Link href={`/product/${product.id}`}>
        <Card.Img src={product.image} variant="top" />
      </Link>

      <Card.Body>
        <Link href={`/product/${product.id}`}>
          <Card.Title as="div">
            <strong>{product.name}</strong>
          </Card.Title>
        </Link>

        <Card.Text as="div">
          <Rating
            value={product.rating}
            text={`${product.num_reviews} reviews`}
          />
        </Card.Text>

        <Card.Text as="h3">${product.price}</Card.Text>
      </Card.Body>
    </Card>
  );
};

export default Product;

File: backend/products/src/update/repo/models/product.py
from pydantic import BaseModel
from typing import Optional

class Product(BaseModel):
    name: Optional[str] = None
    image: Optional[str] = None
    description:Optional[str] = None
    brand:Optional[str] = None
    category: Optional[str] = None
    price: int = 0
    countInStock: int = 0
    rating: int = 0
    numReviews: int = 0